### PR TITLE
First french version with offset management

### PR DESCRIPTION
**Pull request by sigeal**
Hello Radim,
I submit this pull request in case you are interested in integrating the french translation and offset functionality.
I added a couple of strings to the translation (ie. to handle help translation).
Offset functionality has been poorly tested, but will be more in coming weeks, though in the specific case of railways.
Hope you can have a look at this request,
--
Christophe

### CODE REVIEW

**Comment by blazek** [comment]
Thanks for your effort and sorry for my late reply.

There are however many deleted files in the request (RELEASE, README, Makefile...) , I cannot merge something like that and I cannot even test, because ui files cannot be compiled. 

OTOH, there is lrs/deploy.bat  with hardcoded paths and comments in French, which cannot be merged either.

**Comment by sigeal** [comment]
Thank you for your feedback.
I am not familiar with the git PR process with QGIS Plugins.
I will try to make a cleaner PR so you can at least test it.

**Comment by sigeal** [comment]
Added missing files and removed deploy.bat. Couldn't add RELEASE because I work on Windows (I know it's a shame) which doesn't accept a file  with same name as an existing directory (RELEASE and /release).
Is this better for you ?

**Comment by sigeal** [comment]
Most of the modified .py files only have an extra line added at their end, because of my code editor (notepad++). Sorry about that. Actually modified .py files are :
- lrs/lrsbase.py
- lrs/lrsevents.py
- lrs/lrslayerpart.py
- lrs/lrspartbase.py
- lrs/lrsroutebase.py
- ui/utils.py
- ui/lrsdockwidget.py

Compiled ui files ui_lrsdockwidget.py and ui_selectiondialog.py are in /ui folder.

Hope this helps.

**Comment by blazek** [comment]
I think that it should gently handle NULL offset values, I have got 

`  
  File "lrs/lrs/lrslayerpart.py", line 107, in eventSegments
    start = float(start)
TypeError: float() argument must be a string or a number, not 'QVariant'
`

Why lrs/resources.qrc was removed?

I have renamed RELEASE file to avoid conflict on Windows.

**Comment by sigeal** [comment]
Thank you. I added lrs/resources.qrc and RELEASE.txt

Line 107 is about start measure, not start offset. I didn't change this line.
Offset values (oStart and oEnd) have 0.0 default values.

**Comment by blazek** [comment]
> Line 107 is about start measure, not start offset. I didn't change this line.

You are right, sorry. That is my bug which I should fix.

> Offset values (oStart and oEnd) have 0.0 default values.

Function params have defaults, but the data in events table may be null so that it stops with:
`
  File "/home/radim/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lrs/lrs/lrslayerpart.py", line 141, in linestringSegment
    iOffset = (oEnd - oStart) / (toMeasure - fromMeasure)
TypeError: unsupported operand type(s) for -: 'QVariant' and 'QVariant'
`

Another potential problem: (toMeasure - fromMeasure) may be null, division by zero.

BTW, nice feature that proportional offset. I am just curious, what are you using that for in railways.

**Comment by sigeal** [comment]
Thank you for testing and review my code.
I will have a look at issues you raised.

Proportional offset can be use in various contexts, like for fences which are not always parallel to the railway. Also for cables following the railway. In the road context it can be used for guard rails, which are sometime not parallel to the road.

**Comment by sigeal** [comment]
Corrected null offset and (toMeasure - fromMeasure) = 0 issues.

**Comment by blazek** [comment]
Merged, released 1.2.0. Thanks for your contribution.

**Comment by sigeal** [comment]
Thank you for the merge.
It noticed that the i18n directory is not installed when the plugin is loaded from QGIS plugin manager.
Is this intentional ?

**Comment by blazek** [comment]
i18n was missing in Makefile. I have made a new release with fix.

**Comment by sigeal** [comment]
Thank you very much Radim !

File: lrs/lrs/utils.py [hunk 1]
```diff
@@ -175,8 +175,54 @@ def measureAlongPolyline(polyline, segment, pnt):
 
 # layer.dataProvider().deleteFeatures( ids )
 
+# compute offset pt
+def offsetPt(point1, point2, offset):
```
Comment (blazek) [inline lrs/lrs/utils.py:179]: math.fabs(dy) / math.fabs(dx) gives error if dx is zero. 

Should not simply work this?:

a = math.atan2(dy, dx) - math.pi/2
x = point2.x() + offset * math.cos(a)
y = point2.y() + offset * math.sin(a)

**Pull request merged**